core/src/main/scala/org/apache/spark/shuffle/FetchFailedException.scala

```diff
@@ -45,6 +50,12 @@ private[spark] class FetchFailedException(
     this(bmAddress, shuffleId, mapId, reduceId, cause.getMessage, cause)
   }
 
+  // SPARK-19276. We set the fetch failure in the task context, so that even if there is user-code
+  // which intercepts this exception (possibly wrapping it), the Executor can still tell there was
+  // a fetch failure, and send the correct error msg back to the driver.  The TaskContext won't be
+  // defined if this is run on the driver (just in test cases) -- we can safely ignore then.
```

Comment:
This last sentence is confusing.  A task that runs locally on the driver can still hit fetch failures right?  Or are you saying the TaskContext will only be not defined in test cases?